src/ray/raylet/node_manager.cc

```diff
@@ -2766,6 +2769,91 @@ void NodeManager::SendSpilledObjectRestorationRequestToRemoteNode(
       });
 }
 
+void NodeManager::FillNormalTaskResourceUsage(
+    const std::shared_ptr<rpc::ResourcesData> &resources_data) {
+  auto new_resources = GetResourcesUsedByNormalTask();
+  if (new_resources) {
+    std::shared_ptr<std::unordered_map<std::string, double>> resources_changes;
+    if (nullptr == last_report_normal_task_resources_) {
+      resources_changes = new_resources;
+    } else {
+      resources_changes = GetNormalTaskResourcesChanges(
+          *last_report_normal_task_resources_, new_resources);
+    }
+
+    if (!resources_changes->empty()) {
+      for (const auto &iter : *resources_changes) {
+        (*resources_data->mutable_normal_task_resources_changes())[iter.first] =
+            iter.second;
+      }
+      last_report_normal_task_resources_ = new_resources;
+    }
+  }
+}
+
+std::shared_ptr<std::unordered_map<std::string, double>>
+NodeManager::GetResourcesUsedByNormalTask() const {
+  auto resources = std::make_shared<std::unordered_map<std::string, double>>();
+  if (!RayConfig::instance().new_scheduler_enabled()) {
+    return resources;
```

Comment:
Presumably this might lead to incorrect behavior if we turn the new scheduler off right? If so, can we add a warning here?

Reply:
Hmm I think we should just deprecate the old scheduler rather than making it buggy?

Reply:
I have add a warning log here. BTW: @rkooo567 What do you mean of deprecate  the old scheduler?